darts/models/torch_forecasting_model.py

```diff
@@ -71,26 +71,28 @@ def _cat_with_optional(tsr1: torch.Tensor, tsr2: Optional[torch.Tensor]):
     def __len__(self):
         return len(self.ts_dataset)
 
-    def __getitem__(self, idx: int) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+    def __getitem__(self, idx: int) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
         """
         Cast the content of the dataset to torch tensors
         """
         item = self.ts_dataset[idx]
 
-        if len(item) == 2:
+        if isinstance(self.ts_dataset, SimpleInferenceDataset):
             # the dataset contains (input_target, input_covariate) only
-            input_tgt = torch.from_numpy(item[0]).float()
-            input_cov = torch.from_numpy(item[1]).float() if item[1] is not None else None
-            return self._cat_with_optional(input_tgt, input_cov)
+            input_tgt = torch.from_numpy(item[0].values()).float()
+            input_cov = torch.from_numpy(item[1].values()).float() if item[1] is not None else None
+            future_cov = torch.from_numpy(item[2].values()).float() if item[2] is not None else np.array([])
```

Comment:
I can't return `None` here, the data loader will not accept it. But the `np.array([])` workaround is a bit hacky, I'm still thinking about how to improve this. Open to suggestions.

Reply:
Indeed it's a bit hacky. Maybe we need two different Torch Dataset classes?

Reply:
You mean one class that does return future covariates and one that does not?

Reply:
Yes